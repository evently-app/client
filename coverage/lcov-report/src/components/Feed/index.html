<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Feed</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/Feed
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">40.54% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>45/111</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>29/58</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">23.26% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>10/43</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">40.19% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>43/107</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="Filter.js"><a href="Filter.js.html">Filter.js</a></td>
	<td data-value="47.46" class="pic low"><div class="chart"><div class="cover-fill" style="width: 47%;"></div><div class="cover-empty" style="width:53%;"></div></div></td>
	<td data-value="47.46" class="pct low">47.46%</td>
	<td data-value="59" class="abs low">28/59</td>
	<td data-value="61.11" class="pct medium">61.11%</td>
	<td data-value="36" class="abs medium">22/36</td>
	<td data-value="29.17" class="pct low">29.17%</td>
	<td data-value="24" class="abs low">7/24</td>
	<td data-value="47.27" class="pct low">47.27%</td>
	<td data-value="55" class="abs low">26/55</td>
	</tr>

<tr>
	<td class="file low" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="32.69" class="pic low"><div class="chart"><div class="cover-fill" style="width: 32%;"></div><div class="cover-empty" style="width:68%;"></div></div></td>
	<td data-value="32.69" class="pct low">32.69%</td>
	<td data-value="52" class="abs low">17/52</td>
	<td data-value="31.82" class="pct low">31.82%</td>
	<td data-value="22" class="abs low">7/22</td>
	<td data-value="15.79" class="pct low">15.79%</td>
	<td data-value="19" class="abs low">3/19</td>
	<td data-value="32.69" class="pct low">32.69%</td>
	<td data-value="52" class="abs low">17/52</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Apr 14 2019 20:53:00 GMT-0400 (Eastern Daylight Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
<script src="../../../block-navigation.js"></script>
</body>
</html>
