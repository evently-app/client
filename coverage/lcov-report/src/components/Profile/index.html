<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Profile</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/Profile
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">40.35% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>23/57</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">20.83% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>5/24</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">29.41% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/17</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">40.74% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>22/54</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="Dial.js"><a href="Dial.js.html">Dial.js</a></td>
	<td data-value="29.55" class="pic low"><div class="chart"><div class="cover-fill" style="width: 29%;"></div><div class="cover-empty" style="width:71%;"></div></div></td>
	<td data-value="29.55" class="pct low">29.55%</td>
	<td data-value="44" class="abs low">13/44</td>
	<td data-value="11.11" class="pct low">11.11%</td>
	<td data-value="18" class="abs low">2/18</td>
	<td data-value="30.77" class="pct low">30.77%</td>
	<td data-value="13" class="abs low">4/13</td>
	<td data-value="30.95" class="pct low">30.95%</td>
	<td data-value="42" class="abs low">13/42</td>
	</tr>

<tr>
	<td class="file medium" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="76.92" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 76%;"></div><div class="cover-empty" style="width:24%;"></div></div></td>
	<td data-value="76.92" class="pct medium">76.92%</td>
	<td data-value="13" class="abs medium">10/13</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="6" class="abs medium">3/6</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="12" class="abs medium">9/12</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Apr 14 2019 20:53:00 GMT-0400 (Eastern Daylight Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
<script src="../../../block-navigation.js"></script>
</body>
</html>
