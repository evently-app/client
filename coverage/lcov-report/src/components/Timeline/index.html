<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Timeline</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/Timeline
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">12.7% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>16/126</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">10.61% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>7/66</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">2.44% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/41</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">13.01% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>16/123</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="CalendarButton.js"><a href="CalendarButton.js.html">CalendarButton.js</a></td>
	<td data-value="11.11" class="pic low"><div class="chart"><div class="cover-fill" style="width: 11%;"></div><div class="cover-empty" style="width:89%;"></div></div></td>
	<td data-value="11.11" class="pct low">11.11%</td>
	<td data-value="27" class="abs low">3/27</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="14" class="abs low">0/14</td>
	<td data-value="11.11" class="pct low">11.11%</td>
	<td data-value="27" class="abs low">3/27</td>
	</tr>

<tr>
	<td class="file medium" data-value="EventCardPreview.js"><a href="EventCardPreview.js.html">EventCardPreview.js</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="6" class="abs low">1/6</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	</tr>

<tr>
	<td class="file low" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="8.99" class="pic low"><div class="chart"><div class="cover-fill" style="width: 8%;"></div><div class="cover-empty" style="width:92%;"></div></div></td>
	<td data-value="8.99" class="pct low">8.99%</td>
	<td data-value="89" class="abs low">8/89</td>
	<td data-value="5.56" class="pct low">5.56%</td>
	<td data-value="54" class="abs low">3/54</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="21" class="abs low">0/21</td>
	<td data-value="9.3" class="pct low">9.3%</td>
	<td data-value="86" class="abs low">8/86</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Apr 14 2019 20:53:00 GMT-0400 (Eastern Daylight Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
<script src="../../../block-navigation.js"></script>
</body>
</html>
