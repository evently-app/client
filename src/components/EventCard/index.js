import React, { Component } from "react";
import { View, Text } from "react-native";

import { SCREEN_WIDTH, SCREEN_HEIGHT } from "../../lib/constants";

class EventCard extends Component {
	render() {
		return (
			<View
				style={{
					width: SCREEN_WIDTH - 20,
					height: SCREEN_HEIGHT - 100,
					backgroundColor: this.props.backgroundColor,
					borderRadius: 20
				}}
			>
				<Text>{this.props.title}</Text>
			</View>
		);
	}
}

export default EventCard;
