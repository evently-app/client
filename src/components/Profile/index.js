import React, { Component } from "react";
import { View, Text } from "react-native";

import { SCREEN_WIDTH, SCREEN_HEIGHT } from "../../lib/constants";

class Profile extends Component {
	render() {
		return (
			<View
				style={{
					flex: 1,
					width: SCREEN_WIDTH,
					height: SCREEN_HEIGHT,
					backgroundColor: "rgba(0, 255, 0, 0.2)",
					justifyContent: "center",
					alignItems: "center"
				}}
			>
				<Text>Profile</Text>
			</View>
		);
	}
}

export default Profile;
