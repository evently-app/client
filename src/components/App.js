import React, { Component } from "react";
import { Auth } from "../redux/user";
import { connect } from "react-redux";
import { StyleSheet, StatusBar, Animated, View } from "react-native";

import LinearGradient from "react-native-linear-gradient";
import Haptics from "react-native-haptic-feedback";

import Profile from "./Profile";
import Feed from "./Feed";
import Timeline from "./Timeline";
import TouchableScale from "./universal/TouchableScale";

import ProfileLogo from "../assets/profile.svg";
import FeedLogo from "../assets/logo.svg";
import TimelineLogo from "../assets/timeline.svg";

import { SCREEN_WIDTH, SCREEN_HEIGHT } from "../lib/constants";
import { WatchUser } from "../redux/user";

const xOffset = new Animated.Value(SCREEN_WIDTH);
const scrollPosition = Animated.event([{ nativeEvent: { contentOffset: { x: xOffset } } }], {
  useNativeDriver: true
});

const opacityStyle = index => {
  const inputRange = [0, SCREEN_WIDTH, 2 * SCREEN_WIDTH];
  switch (index) {
    case 0:
      return {
        opacity: xOffset.interpolate({
          inputRange,
          outputRange: [1, 0.5, 0.5]
        })
      };
    case 1:
      return {
        opacity: xOffset.interpolate({
          inputRange,
          outputRange: [0.5, 1, 0.5]
        })
      };
    case 2:
      return {
        opacity: xOffset.interpolate({
          inputRange,
          outputRange: [0.5, 0.5, 1]
        })
      };
  }
};

class App extends Component {
  state = {
    currentPage: 1
  };

  componentWillMount() {
    // log in / sign up anonymously
    this.props
      .Auth()
      .then(() => {
        // sync user entity in redux with firestore
        this.watchUser = this.props.WatchUser();
      })
      .catch(error => {
        console.log(error);
      });
  }

  componentDidMount() {
    xOffset.setValue(SCREEN_WIDTH);
    setTimeout(
      () =>
        this.ScrollView.getNode().scrollTo({
          x: SCREEN_WIDTH,
          y: 0,
          animated: false
        }),
      1
    );
  }

  /* navigation functions */
  seeProfile = () => {
    Haptics.trigger("impactMedium");
    this.ScrollView.getNode().scrollTo({ x: 0, y: 0, animated: true });
  };

  seeFeed = () => {
    Haptics.trigger("impactMedium");
    this.ScrollView.getNode().scrollTo({
      x: SCREEN_WIDTH,
      y: 0,
      animated: true
    });
  };

  seeTimeline = () => {
    Haptics.trigger("impactMedium");
    this.ScrollView.getNode().scrollTo({
      x: 2 * SCREEN_WIDTH,
      y: 0,
      animated: true
    });
  };

  handleScrollEnd = ({ nativeEvent }) => {
    const { x } = nativeEvent.contentOffset;
    if (x === 0) this.setState({ currentPage: 0 });
    else if (x === SCREEN_WIDTH) this.setState({ currentPage: 1 });
    else this.setState({ currentPage: 2 });
  };

  render() {
    const { currentPage } = this.state;
    // console.log(this.props);

    return (
      <LinearGradient style={styles.container} locations={[0, 0.9]} colors={["black", "#150218"]}>
        <StatusBar barStyle="light-content" />
        <Animated.ScrollView
          horizontal
          pagingEnabled
          bounces={false}
          ref={ScrollView => (this.ScrollView = ScrollView)}
          // scrollEnabled={currentPage !== 1}
          scrollEnabled={false}
          onMomentumScrollEnd={this.handleScrollEnd}
          showsHorizontalScrollIndicator={false}
          showsVerticalScrollIndicator={false}
          scrollEventThrottle={16}
          onScroll={scrollPosition}
        >
          <Profile />
          <Feed />
          <Timeline />
        </Animated.ScrollView>
        <View style={styles.tabBarContainer}>
          <LinearGradient
            style={{ position: "absolute", bottom: 0, left: 0, right: 0, height: 100 }}
            locations={[0, 0.8, 1]}
            colors={["rgba(21,2,24,0)", "rgba(21,2,24,0.95)", "rgba(21,2,24,1)"]}
          />
          <TouchableScale
            style={styles.tabBarButton}
            animatedStyle={opacityStyle(0)}
            onPress={this.seeProfile}
          >
            <ProfileLogo />
          </TouchableScale>
          <TouchableScale
            style={styles.tabBarButton}
            animatedStyle={opacityStyle(1)}
            onPress={this.seeFeed}
          >
            <FeedLogo />
          </TouchableScale>
          <TouchableScale
            style={styles.tabBarButton}
            animatedStyle={opacityStyle(2)}
            onPress={this.seeTimeline}
          >
            <TimelineLogo />
          </TouchableScale>
        </View>
      </LinearGradient>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  tabBarContainer: {
    position: "absolute",
    bottom: 0,
    left: 0,
    right: 0,
    height: 100,
    padding: 25,
    paddingBottom: 30,
    backgroundColor: "transparent",
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center"
  },
  tabBarButton: {
    width: 60,
    height: 60,
    alignItems: "center",
    justifyContent: "center"
  }
});

const mapStateToProps = state => {
  return {
    uid: state.user.uid
  };
};

const mapDispatchToProps = {
  Auth,
  WatchUser
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
