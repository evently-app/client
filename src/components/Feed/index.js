import React, { Component, Fragment } from "react";
import { Animated, Easing, View, Text, TouchableOpacity, StyleSheet } from "react-native";
import { connect } from "react-redux";
import _ from "lodash";

import Spinner from "../universal/Spinner";
import Filter from "./Filter";
import EventCard from "../EventCard";
import Swipeable from "../EventCard/Swipeable";

import { SCREEN_WIDTH, SCREEN_HEIGHT, SB_HEIGHT } from "../../lib/constants";
import { LoadQueue, UpdateQueue } from "../../redux/queue";
import { SwipeRight, SwipeLeft } from "../../redux/timeline";

class Feed extends Component {
	state = {
		animatedValues: {}
	};

	animatedEntry = new Animated.Value(1);
	filterDrag = new Animated.Value(0);

	componentDidMount() {
		const { LoadQueue, filterTime, filterType } = this.props;
		LoadQueue({ filterTime, filterType }).then(this.entryAnimation);
	}

	static getDerivedStateFromProps(props, state) {
		const {
			UpdateQueue,
			queue,
			currentTypeFilter,
			currentTimeFilter,
			selectedType,
			selectedTime
		} = props;
		const { animatedValues } = state;

		// if there's a mismatch then new events have been loaded
		if (queue.length !== _.size(state.animatedValues)) {
			let newAnimatedValues = {};
			queue.forEach(({ id }) => {
				if (animatedValues[id] == undefined) newAnimatedValues[id] = new Animated.Value(0);
			});

			return { animatedValues: { ...newAnimatedValues, ...animatedValues } };
		}

		// if the filter has changed hide the cards and update the queue
		if (currentTypeFilter !== selectedType || currentTimeFilter != selectedTime) {
			console.log("updating queue");

			this.exitAnimation();
			UpdateQueue({ filterTime: selectedTime, filterType: selectedType }).then(this.entryAnimation);

			return { animatedValues: {} };
		}

		return null;
	}

	entryAnimation = () => {
		Animated.timing(this.animatedEntry, {
			toValue: 0,
			duration: 300,
			easing: Easing.quad,
			useNativeDriver: true
		}).start();
	};

	exitAnimation = () => {
		Animated.timing(this.animatedEntry, {
			toValue: 1,
			duration: 300,
			easing: Easing.quad,
			useNativeDriver: true
		}).start();
	};

	popCard = ({ id }) => {
		const { UpdateQueue, filterTime, filterType } = this.props;
		const { animatedValues } = this.state;

		// if fewer than five events remaining in the queue, reload
		if (_.size(animatedValues) < 5) UpdateQueue({ filterTime, filterType });

		this.setState({
			animatedValues: _.omit(animatedValues, id)
		});
	};

	onSwipeCardRight = card => {
		this.props.SwipeRight(card);
		this.popCard(card);
	};

	onSwipeCardLeft = card => {
		this.props.SwipeLeft(card);
		this.popCard(card);
	};

	handleOnStartSwipe = () => {
		// this.fetchCards();
	};

	render() {
		const { animatedValues } = this.state;
		const { queue, loading, userLocation } = this.props;

		const cardContainerStyle = {
			opacity: this.animatedEntry.interpolate({
				inputRange: [0, 1],
				outputRange: [1, 0]
			}),
			transform: [
				{
					translateY: this.filterDrag.interpolate({
						inputRange: [0, 100],
						outputRange: [0, 100]
					})
				},
				{
					scale: this.animatedEntry.interpolate({
						inputRange: [0, 1],
						outputRange: [1, 0.9]
					})
				}
			]
		};

		const first = queue.length - 1;
		const cards = (
			<Animated.View style={[styles.center, cardContainerStyle]}>
				{queue.map((card, i) => (
					<Swipeable
						key={card.id}
						id={card.id}
						index={queue.length - i}
						swipeAmount={animatedValues[card.id]}
						scaleAmount={i !== first ? animatedValues[queue[i + 1].id] : null}
						onStartSwipe={this.handleOnStartSwipe}
						onSwipeRight={() => this.onSwipeCardRight(card)}
						onSwipeLeft={() => this.onSwipeCardLeft(card)}
					>
						<EventCard userLocation={userLocation} {...card} />
					</Swipeable>
				))}
			</Animated.View>
		);

		return (
			<View style={styles.container}>
				{loading ? <Spinner /> : cards}
				<Filter filterDrag={this.filterDrag} />
			</View>
		);
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		width: SCREEN_WIDTH,
		height: SCREEN_HEIGHT,
		overflow: "hidden",
		alignItems: "center",
		justifyContent: "center"
	},
	center: {
		alignItems: "center",
		justifyContent: "center"
	}
});

const mapStateToProps = ({ queue, filter, user }) => {
	return {
		queue: queue.queue,
		currentTypeFilter: queue.currentTypeFilter,
		currentTimeFilter: queue.currentTimeFilter,
		loading: queue.isLoadingQueue,
		selectedTime: filter.selectedTime,
		selectedType: filter.selectedType,
		userLocation: user.location
	};
};

const mapDispatchToProps = {
	LoadQueue,
	UpdateQueue,
	SwipeRight,
	SwipeLeft
};

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(Feed);
