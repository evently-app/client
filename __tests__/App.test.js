import "react-native";
import React from "react";
import { App } from "../src/components/App";

jest.mock("react-native-code-push");

it("App renders correctly", () => {
	const snap = shallow(
		<App
			Auth={() => {
				return Promise.resolve();
			}}
			WatchUser={() => {
				return () => {};
			}}
		/>
	);
	expect(snap).toMatchSnapshot();
});
