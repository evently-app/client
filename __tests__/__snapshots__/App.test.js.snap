// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<LinearGradient
  colors={
    Array [
      "black",
      "#150218",
    ]
  }
  end={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={
    Array [
      0,
      0.9,
    ]
  }
  start={
    Object {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <StatusBar
    animated={false}
    barStyle="light-content"
    showHideTransition="fade"
  />
  <WrappedAnimatedInteractableView
    animatedNativeDriver={true}
    animatedValueX={-750}
    dragEnabled={false}
    horizontalOnly={true}
    initialPosition={
      Object {
        "damping": 0.5,
        "tension": 700,
        "x": -750,
      }
    }
    snapPoints={
      Array [
        Object {
          "damping": 0.5,
          "tension": 700,
          "x": 0,
        },
        Object {
          "damping": 0.5,
          "tension": 700,
          "x": -750,
        },
        Object {
          "damping": 0.5,
          "tension": 700,
          "x": -1500,
        },
      ]
    }
    style={
      Object {
        "flexDirection": "row",
        "height": 1334,
        "position": "absolute",
        "width": 2250,
      }
    }
  >
    <Connect(Profile) />
    <Connect(Feed) />
    <Connect(Timeline) />
  </WrappedAnimatedInteractableView>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "transparent",
        "bottom": 0,
        "flexDirection": "row",
        "height": 75,
        "justifyContent": "space-between",
        "left": 0,
        "padding": 15,
        "paddingBottom": 15,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <LinearGradient
      colors={
        Array [
          "rgba(21,2,24,0)",
          "rgba(21,2,24,0.95)",
          "rgba(21,2,24,1)",
        ]
      }
      end={
        Object {
          "x": 0.5,
          "y": 1,
        }
      }
      locations={
        Array [
          0,
          0.8,
          1,
        ]
      }
      start={
        Object {
          "x": 0.5,
          "y": 0,
        }
      }
      style={
        Object {
          "bottom": 0,
          "height": 100,
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    />
    <TouchableScale
      animatedStyle={
        Object {
          "opacity": 0.5,
        }
      }
      key="0"
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 60,
          "justifyContent": "center",
          "width": 60,
        }
      }
    >
      <[object Object] />
    </TouchableScale>
    <TouchableScale
      animatedStyle={
        Object {
          "opacity": 1,
        }
      }
      key="1"
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 60,
          "justifyContent": "center",
          "width": 60,
        }
      }
    >
      <[object Object] />
    </TouchableScale>
    <TouchableScale
      animatedStyle={
        Object {
          "opacity": 0.5,
        }
      }
      key="2"
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 60,
          "justifyContent": "center",
          "width": 60,
        }
      }
    >
      <[object Object] />
    </TouchableScale>
  </View>
</LinearGradient>
`;
